okhttp/src/main/java/okhttp3/internal/platform/android/AndroidCertificateChainCleaner.kt

```diff
@@ -48,7 +50,8 @@ internal class AndroidCertificateChainCleaner(
   }
 
   override fun equals(other: Any?): Boolean =
-      other is AndroidCertificateChainCleaner // All instances are equivalent.
+      other is AndroidCertificateChainCleaner &&
+          other.trustManager === this.trustManager
```

Comment:
This could break connection pooling for some clients, but hopefully an edge case.